google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerCloudMonitoringExporterUtils.java

```diff
@@ -0,0 +1,211 @@
+import io.opentelemetry.sdk.metrics.data.MetricDataType;
+import io.opentelemetry.sdk.metrics.data.PointData;
+import io.opentelemetry.sdk.metrics.data.SumData;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+class SpannerCloudMonitoringExporterUtils {
+
+  private static final Logger logger =
+      Logger.getLogger(SpannerCloudMonitoringExporterUtils.class.getName());
+
+  private SpannerCloudMonitoringExporterUtils() {}
+
+  static String getProjectId(PointData pointData) {
+    return pointData.getAttributes().get(PROJECT_ID_KEY);
+  }
+
+  static List<TimeSeries> convertToSpannerTimeSeries(List<MetricData> collection) {
+    List<TimeSeries> allTimeSeries = new ArrayList<>();
+
+    for (MetricData metricData : collection) {
+      // Get common metrics data from GAX library
+      if (!metricData.getInstrumentationScopeInfo().getName().equals(GAX_METER_NAME)) {
+        // Filter out metric data for instruments that are not part of the spanner metrics list
+        continue;
+      }
+      metricData.getData().getPoints().stream()
+          .map(pointData -> convertPointToSpannerTimeSeries(metricData, pointData))
+          .forEach(allTimeSeries::add);
+    }
+
+    return allTimeSeries;
+  }
+
+  private static TimeSeries convertPointToSpannerTimeSeries(
+      MetricData metricData, PointData pointData) {
+    TimeSeries.Builder builder =
+        TimeSeries.newBuilder()
+            .setMetricKind(convertMetricKind(metricData))
+            .setValueType(convertValueType(metricData.getType()));
+    Metric.Builder metricBuilder = Metric.newBuilder().setType(metricData.getName());
+
+    Attributes attributes = pointData.getAttributes();
+    MonitoredResource.Builder monitoredResourceBuilder =
+        MonitoredResource.newBuilder().setType(SPANNER_RESOURCE_TYPE);
+
+    for (AttributeKey<?> key : attributes.asMap().keySet()) {
+      if (SPANNER_PROMOTED_RESOURCE_LABELS.contains(key)) {
+        monitoredResourceBuilder.putLabels(key.getKey(), String.valueOf(attributes.get(key)));
+      } else {
+        metricBuilder.putLabels(key.getKey(), String.valueOf(attributes.get(key)));
+      }
+    }
+
+    builder.setResource(monitoredResourceBuilder.build());
+    builder.setMetric(metricBuilder.build());
+
+    TimeInterval timeInterval =
+        TimeInterval.newBuilder()
+            .setStartTime(Timestamps.fromNanos(pointData.getStartEpochNanos()))
+            .setEndTime(Timestamps.fromNanos(pointData.getEpochNanos()))
+            .build();
+
+    builder.addPoints(createPoint(metricData.getType(), pointData, timeInterval));
+
+    return builder.build();
+  }
+
+  private static MetricKind convertMetricKind(MetricData metricData) {
+    switch (metricData.getType()) {
+      case HISTOGRAM:
+      case EXPONENTIAL_HISTOGRAM:
+        return convertHistogramType(metricData.getHistogramData());
+      case LONG_GAUGE:
```

Comment:
Ditto about not allowing fall through and instead putting similar case values on the same line.